apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerClientPartial.RequestIdSource.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Threading;
+
+namespace Google.Cloud.Spanner.V1;
+
+public partial class SpannerClientImpl
+{
+    internal static ulong ProcessId
+    {
+        set => RequestIdSource.ProcessId = value;
+    }
+
+    /// <summary>
+    /// Manages the generation of unique <see cref="RequestId"/> instances for a specific
+    /// <see cref="SpannerClient"/> instance.
+    /// </summary>
+    /// <remarks>
+    /// This class acts as a dedicated factory for <see cref="RequestId"/> instances.
+    /// There should be a 1:1 mapping between a <see cref="SpannerClient"/> instance and a <see cref="RequestIdSource"/> instance,
+    /// which simplifies the <see cref="SpannerClient"/> by centralizing the request ID generation logic.
+    /// </remarks>
+    private sealed class RequestIdSource
+    {
+        private static ulong s_processId;
+        private static bool s_isProcessIdSet = false;
+        private static readonly object s_lock = new object();
+
+        /// <summary>
+        /// A unique ID for the user application process. This value can only be set once.
```

Comment:
OK, I think this is a good compromise, that it can only be set once. But then this can be simplified quite a lot I think and we don't need the lock or the bool, etc.

- On one side let's have `ulong? userProvidedProcessId` with accompanying property. That can be changed with Interlocked.ComparedExchange only if the current value is null, throws otherwise.
- On the other side let's have `Lazy<ulong> EffectiveProcessId`. The lazy intializar changes `userProvidedProcessId` to 0 if it's null using also interlocked.CompareExchange (it can't be changed anymore and we get the possible user set value if any). If there was a user value the lazy intializer returns that otherwise it creates the random value.
- We only access the EffectiveProcessId.